src/index.ts

```diff
@@ -0,0 +1,257 @@
+      resource: this.resource,
+      timestamp: record.time,
+      // BUNYAN_TO_STACKDRIVER does not have index signature.
+      // tslint:disable-next-line:no-any
+      severity: (BUNYAN_TO_STACKDRIVER as any)[record.level as string],
+    };
+
+    // If the record contains a httpRequest property, provide it on the entry
+    // metadata. This allows Stackdriver to use request log formatting.
+    // https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#HttpRequest
+    // Note that the httpRequest field must properly validate as a HttpRequest
+    // proto message, or the log entry would be rejected by the API. We do no
+    // validation here.
+    if (record.httpRequest) {
+      entryMetadata.httpRequest = record.httpRequest;
+      delete record.httpRequest;
+    }
+
+    // record does not have index signature.
+    // tslint:disable-next-line:no-any
+    if ((record as any)[LOGGING_TRACE_KEY]) {
+      // tslint:disable-next-line:no-any
+      entryMetadata.trace = (record as any)[LOGGING_TRACE_KEY];
+      // tslint:disable-next-line:no-any
+      delete (record as any)[LOGGING_TRACE_KEY];
+    }
+
+    return this.stackdriverLog.entry(entryMetadata, record);
+  }
+
+  /**
+   * Intercept log entries as they are written so we can attempt to add the
+   * trace ID in the same continuation as the function that wrote the log,
+   * because the trace agent currently uses continuation local storage for the
+   * trace context.
+   *
+   * By the time the Writable stream buffer gets flushed and _write gets called
+   * we may well be in a different continuation.
+   */
+  // Writable.write used 'any' in function signature.
+  // tslint:disable-next-line:no-any
+  write(record: any, callback?: Function): boolean;
+  // tslint:disable-next-line:no-any
+  write(record: any, encoding?: string, callback?: Function): boolean;
+  // tslint:disable-next-line:no-any
+  write(...args: any[]): boolean {
+    let record = args[0];
+    let encoding: string|null = null;
+    let callback: Function;
+    if (typeof args[1] === 'string') {
+      encoding = args[1];
+      callback = args[2];
+    } else {
+      callback = args[1];
+    }
+    record = extend({}, record);
+    // record does not have index signature.
+    // tslint:disable-next-line:no-any
+    if (!(record as any)[LOGGING_TRACE_KEY]) {
+      const trace = getCurrentTraceFromAgent();
+      if (trace) {
+        // tslint:disable-next-line:no-any
+        (record as any)[LOGGING_TRACE_KEY] = trace;
+      }
+    }
+    if (encoding !== null) {
+      return super.write.call(this, record, encoding, callback);
+    } else {
+      return super.write.call(this, record, callback);
+    }
+  }
+
+  /**
+   * Relay a log entry to the logging agent. This is called by bunyan through
+   * Writable#write.
+   *
+   * @param {object} record - Bunyan log record.
+   *
+   * @private
+   */
+  _write(record: BunyanLogRecord, encoding?: string, callback?: Function) {
```

Comment:
Can we be more specific than `Function`?

Reply:
Same as write.